iceoryx_binding_c/source/cpp2c_enum_translation.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_binding_c/internal/cpp2c_enum_translation.hpp"
+
+using namespace iox;
+using namespace iox::popo;
+
+namespace cpp2c
+{
+iox_SubscribeState SubscribeState(const iox::SubscribeState value)
+{
+    switch (value)
+    {
+    case SubscribeState::NOT_SUBSCRIBED:
+        return iox_SubscribeState::SubscribeState_NOT_SUBSCRIBED;
```

Comment:
As fas as I remember QAC will complain about missing breaks and the return statements inside the cases.

Reply:
Really? Why? It does not make any sense ...

Reply:
Another big counter argument is: If we would like to have a 100% test coverage and are not allowed to ship unreachable code we definitely cannot write a break after return since we could never reach that code.

Reply:
you could create a `auto retVal =  iox_SubscribeState::SubscribeState_UNDEFINED_ERROR` before the switch and just return that after the switch. what do you think?

Reply:
Why do I have to add complexity to the code and make it more error prone to fix a QAC warning? Here I would recommend the following approach: We do not write an exception for the rule and we do not fix this problem by adding 100 lines of new code - by adding additional breaks, a variable and a return at the end (there are multiple functions which have the same problem here).

We adjust the QAC rule and say: either break or a return. I think the argument is valid and if @elBoberido @marthtz @mossmaurice @budrus @dkroenke  would support me on this I would start this endeavor!

Reply:
Early return in a case is fine and potentially more efficient (no additional jump needed as with a break only to return then). It is also perfectly safe (as long all cleanup is performed on each return points if needed.)

Also, does this mean a fallthrough logic to the next case without break is not allowed? This is sometimes useful as well. No hard rules such as always break in each case please...

Reply:
> 
> 
> you could create a `auto retVal = iox_SubscribeState::SubscribeState_UNDEFINED_ERROR` before the switch and just return that after the switch. what do you think?

That's QAC preferred way I guess.

Reply:
> fallthrough logic to the next case without break is not allowed

Doesn't a fallthrough also generate a compiler warning without an extra attribute? Maybe QAC is fine if the attribute is given. 
However `[[fallthrough]]` is C++17 so I don't think we can use fallthough at all.